irohad/torii/processor/impl/query_processor_impl.cpp

```diff
@@ -50,31 +41,18 @@ namespace iroha {
             };
     }
 
-    rxcpp::observable<
-        std::shared_ptr<shared_model::interface::BlockQueryResponse>>
+    iroha::expected::Result<void, std::string>
     QueryProcessorImpl::blocksQueryHandle(
         const shared_model::interface::BlocksQuery &qry) {
-      using shared_model::interface::BlockQueryResponse;
-      auto make_error = [this](std::string &&error)
-          -> rxcpp::observable<std::shared_ptr<BlockQueryResponse>> {
-        std::shared_ptr<BlockQueryResponse> response =
-            response_factory_->createBlockQueryResponse(std::move(error));
-        return rxcpp::observable<>::just(std::move(response));
-      };
-
-      return qry_exec_
-          ->createQueryExecutor(pending_transactions_, response_factory_)
-          .match(
-              [&](const auto &executor) {
-                if (executor.value->validate(qry, true)) {
-                  return blocks_query_subject_.get_observable();
-                }
-                return make_error("stateful invalid");
-              },
-              [&](const auto &e) {
-                log_->error("Could not validate query: {}", e.error);
-                return make_error("Internal error during query validation.");
-              });
+      auto maybe_query_executor = qry_exec_->createQueryExecutor(
+          pending_transactions_, response_factory_);
```

Comment:
Can we avoid ```QueryExecutor``` recreation each time?

Reply:
I don't think so, we need instances with separate sessions for the clients.